ARM-template/CreateUiDefinition.json

```diff
@@ -0,0 +1,648 @@
+                "label": "User name",
+                "defaultValue": "",
+                "toolTip": "Specify a username for accessing the controller node of the cluster. Must be a valid Ubuntu username",
+                "constraints": {
+                    "required": true,
+                    "regex": "^(?!(?:adm|admin|audio|backup|bin|cdrom|crontab|daemon|dialout|dip|disk|fax|floppy|fuse|games|gnats|irc|kmem|landscape|libuuid|list|lp|mail|man|messagebus|mlocate|netdev|news|nobody|nogroup|operator|plugdev|proxy|root|sasl|shadow|src|ssh|sshd|staff|sudo|sync|sys|syslog|tape|tty|users|utmp|uucp|video|voice|whoopsie|www\\-data)$)\\w+$",
+                    "validationMessage": "Username must not be a reserved Ubuntu username"
+                },
+                "osPlatform": "Linux",
+                "visible": true
+            },
+            {
+                "name": "password",
+                "type": "Microsoft.Common.PasswordBox",
+                "label": {
+                    "password": "Password",
+                    "confirmPassword": "Confirm Password"
+                },
+                "toolTip": "Default password for Kubernetes cluster portal, controllor node, Kibana, etc",
+                "constraints": {
+                    "required": true,
+                    "regex": "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-\\.]).{8,}$",
+                    "validationMessage": "Password must at least 8 characters long and contain lower case, upper case, digit and special character."
+                },
+                "visible": true
+            }
+        ],
+        "steps": [
+            {
+                "name": "commonSettings",
+                "label": "Common Settings",
+                "subLabel": {
+                    "preValidation": "Required",
+                    "postValidation": "Done"
+                },
+                "bladeTitle": "Common Settings",
+                "elements": [
+                    {
+                        "name": "dnsNamePrefix",
+                        "type": "Microsoft.Common.TextBox",
+                        "label": "Dns name prefix",
+                        "defaultValue": "",
+                        "toolTip": "Dns prefix for all components, some random value will be appended to be unique",
+                        "constraints": {
+                            "required": true
+                        },
+                        "visible": true
+                    }
+                ]
+            },
+            {
+                "name": "acsSettings",
+                "label": "ACS Settings",
+                "subLabel": {
+                    "preValidation": "Required",
+                    "postValidation": "Done"
+                },
+                "bladeTitle": "ACS Settings",
+                "elements": [
+                    {
+                        "name": "agentCount",
+                        "type": "Microsoft.Common.DropDown",
```

Comment:
how about making it a textbox with validation? 100 items dropdown sounds crazy:)